ui/src/contexts/Application.jsx

```diff
@@ -86,10 +84,32 @@ function watchVault(id, dispatch, offerStatus) {
     window.localStorage.setItem(id, VaultStatus.CLOSED);
   }
 
-  vaultUpdater().catch(err => {
-    console.error('Vault watcher exception', id, err);
-    dispatch(updateVault({ id, vault: { status: VaultStatus.ERROR, err } }));
-  });
+  async function assetUpdater(asset) {
+    for await (const value of iterateNotifier(asset)) {
+      console.log('======== ASSET', id, value);
+      dispatch(
+        updateVault({
+          id,
+          vault: { asset: { ...value } },
+        }),
+      );
+    }
+  }
+
+  async function watch() {
+    const { vault, asset } = await E(walletP).getPublicNotifiers(id);
+
+    assetUpdater(asset).catch(err => {
+      console.error('Asset watcher exception', id, err);
```

Comment:
is this error reporting sufficient? why not dispatch `VaultStatus.ERROR`?

Reply:
This was just a judgement call, I'm not entirely certain. Error is treated like a terminal state and you can't interact with the vault anymore, so I didn't think we'd want the asset notifier alone to be able to break the entire vault.